plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestPlugin.java

```diff
@@ -95,27 +99,37 @@ public String description() {
     public Settings additionalSettings() {
         return settingsBuilder()
                 .put(PipelineExecutionService.additionalSettings(nodeSettings))
-                // TODO: in a followup issue this should be made configurable
-                .put(NODE_INGEST_SETTING, true)
                 .build();
     }
 
     public void onModule(ActionModule module) {
         if (transportClient == false) {
-            module.registerFilter(IngestActionFilter.class);
+            if (ingestEnabled) {
+                module.registerFilter(IngestActionFilter.class);
+            } else {
+                module.registerFilter(IngestDisabledActionFilter.class);
+            }
+        }
+        if (ingestEnabled) {
+            module.registerAction(PutPipelineAction.INSTANCE, PutPipelineTransportAction.class);
+            module.registerAction(GetPipelineAction.INSTANCE, GetPipelineTransportAction.class);
+            module.registerAction(DeletePipelineAction.INSTANCE, DeletePipelineTransportAction.class);
+            module.registerAction(SimulatePipelineAction.INSTANCE, SimulatePipelineTransportAction.class);
```

Comment:
as I said below on the REST action, I don't know if this is right. I think we should still register these CRUD actions, but don't register simulate. I think it all depends on the meaning that we give to the ingest flag. does it mean that the plugin is disabled completely or that data ingestion (pipeline/processors) is disabled? I think the latter...